lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1791,6 +1793,56 @@ class DecomposeAtenToDtypeLayoutOp
 };
 } // namespace
 
+namespace {
+// Decompose the `aten.select_scatter` operation into `aten.slice_scatter` op.
+class DecomposeAtenSelectScatterOp
+    : public OpRewritePattern<AtenSelectScatterOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenSelectScatterOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value start = op.index();
+    Value dim = op.dim();
+    Value self = op.self();
+    Value src = op.src();
+
+    Value one =
+        rewriter.create<ConstantIntOp>(loc, rewriter.getI64IntegerAttr(1));
+    Value startPlusOne =
+        rewriter.create<AtenAddIntOp>(loc, one.getType(), start, one);
+    BaseTensorType srcTensorType = src.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> sizes;
+    int64_t dimInt = 0;
+    if (srcTensorType.hasSizes()) {
+      ArrayRef<int64_t> srcShape = srcTensorType.getSizes();
+      // `src` has a reduced rank. Hence add 1.
+      int64_t srcRank = srcShape.size() + 1;
+      if (matchPattern(dim, m_TorchConstantInt(&dimInt))) {
+        dimInt = toPositiveDim(dimInt, srcRank);
+        if (!isValidDim(dimInt, srcRank))
+          return rewriter.notifyMatchFailure(op, "dim is not a valid dim");
+
+        sizes.append(srcShape.begin(), srcShape.end());
+        sizes.insert(sizes.begin() + dimInt, 1);
+
+      } else {
+        sizes.resize(srcShape.size() + 1, kUnknownSize);
```

Comment:
what is the reason for adding 1 here?

Reply:
Actually according to the `select_scatter` the selection dimension is squeezed out. But in the `slice_scatter` we need the selection/slicing dimension(The reason of adding unsqueezeOp).